fedimint-server/src/consensus/mod.rs

```diff
@@ -406,9 +394,8 @@ impl FedimintConsensus {
             let span = info_span!("Processing transaction");
             async {
                 trace!(?transaction);
-                dbtx.remove_entry(&ProposedTransactionKey(txid))
-                    .await
-                    .expect("DB Error");
+                let mut tx_cache = self.tx_cache.lock().unwrap();
```

Comment:
Holding a blocking mutex across `.await`. :fearful: 

We're lucky that compiler complained about it when I was rebasing #1599

Reply:
Not sure what the alternative is though in this case...

Reply:
tokio mutex?

Reply:
Nothing comes to my mind that would be better than the existing Rust lints. I discovered this because the whole future now needs to be `Send`, etc. Seems like not an easy problem to solve.